### PR TITLE
fix: failFast not considered in DAG

### PR DESCRIPTION
**Pull request by simster7**
Fixes: https://github.com/argoproj/argo-workflows/issues/5223

This change is in a high-impact part of the code, so I will spend time writing tests before making this ready for review

### CODE REVIEW

File: examples/dag-disable-failFast.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ spec:
       args: ["hello world"]
   - name: statis
     dag:
-      failFast: false
+      failFast: true
```
Comment (simster7) [inline examples/dag-disable-failFast.yaml:34]: Revert before merge

**Comment by simster7** [comment]
Seems this is only relevant for an edge case: if there is a node running while a failFast: true DAG has failed, as long as a node is running new nodes will be able to be created

**Comment by swathi59** [comment]
Hi simster7,
Thank you for the updates, could you let me know when will be the fix available.

**Comment by simster7** [comment]
Hi @swathi59, we decided to not fix this for now and instead iterate on a potential new feature: https://github.com/argoproj/argo-workflows/issues/5398. The behavior you experienced has been in the code base since `failFast` was introduced and we're not sure we want to "fix" it

**Pull request closed without merging**